packages/accounts-password/password_server.js

```diff
@@ -239,8 +239,18 @@ const checkForCaseInsensitiveDuplicates = (fieldName, displayName, fieldValue, o
   const skipCheck = Object.prototype.hasOwnProperty.call(Accounts._skipCaseInsensitiveChecksForTest, fieldValue);
 
   if (fieldValue && !skipCheck) {
+    let selector = {};
+    const { useLowerCaseEmails } = Accounts._options;
+    // if useLowerCaseEmails is true in Accounts.config and fieldName='emails.address'
+    // then don't run selectorForFastCaseInsensitiveLookup
+    // instead run a simple regex query for email.
+    if (useLowerCaseEmails && (fieldName === 'emails.address')) {
+      selector[fieldName] = new RegExp(`^${Meteor._escapeRegExp(fieldValue)}$`, 'i');
```

Comment:
Why not just use MongoDB regex option?
https://docs.mongodb.com/manual/reference/operator/query/regex/